app/models/supermarket.rb

```diff
@@ -1,3 +1,9 @@
 class Supermarket < ApplicationRecord
   has_many :items
+  has_many :customer_items, through: :items
+  has_many :customers, through: :customer_items
```

Comment:
Nice use of the additional associations! It's also okay to skip like 3 and just say `has_many :customers, through: :items` since a direct association between items and customers is already set up